### PR TITLE
reference: fix issues in java best practice doc

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

Fix issues in java best practice doc

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1884

### Which version does your change affect? 

dev, v3.0, v2.1

### CODE REVIEW

File: dev/reference/best-practices/java-app.md [hunk 1]
```diff
@@ -302,10 +296,10 @@ jstack 对应于 Go 中的 pprof/goroutine，可以比较方便地排查进程
 
 #### 火焰图
 
-Java 应用中获取火焰图较繁琐，可以通过[这样](http://psy-lob-saw.blogspot.com/2017/02/flamegraphs-intro-fire-for-everyone.html)手工获取。
+Java 应用中获取火焰图较繁琐，可参阅[Java Flame Graphs Introduction: Fire For Everyone!](http://psy-lob-saw.blogspot.com/2017/02/flamegraphs-intro-fire-for-everyone.html)来手动获取。
```
Comment (yikeke) [inline dev/reference/best-practices/java-app.md]:
```suggestion
Java 应用中获取火焰图较繁琐，可参阅 [Java Flame Graphs Introduction: Fire For Everyone!](http://psy-lob-saw.blogspot.com/2017/02/flamegraphs-intro-fire-for-everyone.html) 来手动获取。
```

**Comment by yikeke** [comment]
Feel free to leave comments. @lilin90 @lysu

File: dev/TOC.md [hunk 2]
```diff
@@ -131,7 +131,7 @@
     - [工具下载](/dev/reference/tools/download.md)
   + 最佳实践
     - [HAProxy 最佳实践](/dev/reference/best-practices/haproxy.md)
-    - [Java 使用 TiDB 的最佳实践](/dev/reference/best-practices/using-tidb-in-java.md)
+    - [Java 应用使用 TiDB 的最佳实践](/dev/reference/best-practices/java-app.md)
```
Comment (lilin90) [inline dev/TOC.md]:
```suggestion
    - [Java 应用开发最佳实践](/dev/reference/best-practices/java-app.md)
```

File: dev/reference/best-practices/java-app.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
 ---
-title: Java 使用 TiDB 的最佳实践
+title: Java 应用使用 TiDB 的最佳实践
```
Comment (lilin90) [inline dev/reference/best-practices/java-app.md]:
```suggestion
title: 开发 Java 应用使用 TiDB 的最佳实践
```

File: dev/reference/best-practices/java-app.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
 category: reference
 ---
 
-# Java 使用 TiDB 的最佳实践
+# Java 应用使用 TiDB 的最佳实践
```
Comment (lilin90) [inline dev/reference/best-practices/java-app.md]:
```suggestion
# 开发 Java 应用使用 TiDB 的最佳实践
```

File: dev/reference/best-practices/java-app.md [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
 
-本文主要介绍基于 TiDB 开发 Java 应用程序时可能需要关注的问题。
+本文主要介绍开发 Java 应用程序时使用 TiDB 可能需要关注的问题。
```
Comment (lilin90) [inline dev/reference/best-practices/java-app.md]:
```suggestion
本文主要介绍开发 Java 应用程序来使用 TiDB 时需要关注的问题与最佳实践。
```

**Comment by lilin90** [comment]
@yikeke I think you can update two other versions now.

File: dev/reference/best-practices/java-app.md [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
-title: Java 使用 TiDB 的最佳实践
+title: 开发 Java 应用使用 TiDB 的最佳实践
 category: reference
 ---
 
-# Java 使用 TiDB 的最佳实践
+# 开发 Java 应用使用 TiDB 的最佳实践
 
-本文主要介绍基于 TiDB 开发 Java 应用程序时可能需要关注的问题。
+本文主要介绍开发 Java 应用程序来使用 TiDB 时需要关注的问题与最佳实践。
```
Comment (yikeke) [inline dev/reference/best-practices/java-app.md]:
```suggestion
本文主要介绍开发 Java 应用程序时使用 TiDB 的常见问题与最佳实践。
```
为了和最后一段统一说法，这里改成这样可以不？ @lilin90

Comment (lilin90) [reply dev/reference/best-practices/java-app.md]: 这样呢？ @yikeke 
```suggestion
本文主要介绍如何开发 Java 应用程序以更好地使用 TiDB，包括开发中的常见问题与最佳实践。
```

**Pull request merged**